src/api/firebase.js

```diff
@@ -72,11 +76,41 @@ export async function addItem(listId, { itemName, daysUntilNextPurchase }) {
 		totalPurchases: 0,
 	});
 }
+/**
+ * Update an existing item in the user's list in Firestore.
+ * @param {string} listId The id of the list we're adding to.
+ * @param {Object} item Information about the new item.
+ **/
+export async function updateItem(listId, item) {
+	const {
+		id,
+		dateCreated,
+		dateLastPurchased,
+		dateNextPurchased,
+		totalPurchases,
+	} = item;
+
+	const transformToJSDate = (date) => {
+		return date.toDate();
+	};
+	const prevEstimate = getDaysBetweenDates(
+		transformToJSDate(dateCreated),
+		transformToJSDate(dateNextPurchased),
+	);
+	const daysSinceLastPurchase = getDaysBetweenDates(
+		transformToJSDate(dateLastPurchased ? dateLastPurchased : dateCreated),
+		new Date(),
+	);
 
-export async function updateItem(listId, itemId) {
-	const itemRef = doc(db, listId, itemId);
+	const estimatedNextPurchase = calculateEstimate(
```

Comment:
The acceptance criteria require the dateNextPurchased to be a date and not a whole number returned from calculateEstimate. Therefore we need to store it in a variable to be used in getNextPurchaseDate, which returns a date.